lib/galaxy/jobs/dynamic_tool_destination.py

```diff
@@ -1416,14 +1591,183 @@ def map_tool_to_destination(
     return destination
 
 
+def get_destination_list_from_job_config(job_config_location):
```

Comment:
Hehe, because I didn't know it existed. Perhaps I should've done a bit more research. I'll try the built-in parser!

Reply:
Would you happen to know if it's possible to grab all destinations using [`get_destinations`](https://github.com/galaxyproject/galaxy/blob/6d94009cb359ef58acd987a0fbdf3f9d024c7c17/lib/galaxy/jobs/__init__.py#L551), like by passing in an empty string or wildcard or something? I guess the other option would be throwing out the `destination_list` completely and checking if destinations are valid on the fly with [`get_destination`](https://github.com/galaxyproject/galaxy/blob/6d94009cb359ef58acd987a0fbdf3f9d024c7c17/lib/galaxy/jobs/__init__.py#L536).

Reply:
After some more digging, it looks like [`get_destinations`](https://github.com/galaxyproject/galaxy/blob/6d94009cb359ef58acd987a0fbdf3f9d024c7c17/lib/galaxy/jobs/__init__.py#L551) cannot be run outside the core galaxy code, and I don't see any equivalent functions in [galaxy-lib](https://github.com/galaxyproject/galaxy-lib). Because of this, I believe parsing `job_conf.xml` inside DTD is the best option, as it allows users to validate configurations with destination-checking by running the file directly, in addition to having destination-checking occurring at Galaxy runtime.

Reply:
Ok, so when the destination is running you can use app.job_config.get_destination(destination), but in offline validation mode you do not have access to the app object. One potential solution is to use your parser optionally in offline mode so that if the job_config.xml get's changed in the future your code won't break. And when in online mode use the app object to get_destination. Though I will defer to @Takadonet for his opinion.

Reply:
Think the best options would be to only use your parser when in offline mode. So if job_conf.xml changes, only validation checking is broken and not 'online' mode.

Reply:
Sounds like a solid plan!